ReflectometryServer/test_modules/test_beamline_parameter.py

```diff
@@ -463,6 +464,21 @@ def test_GIVEN_three_beamline_parameters_in_mode_WHEN_1st_changed_and_move_beaml
 
         assert_that(moves, contains(1, 1, 1), "beamline parameter move counts")
 
+    @parameterized.expand([("s1", 12.132, "theta", 4.012), ("s3", 1.123, "theta", 2.342)])
+    def test_GIVEN_parameter_with_new_sp_WHEN_theta_moved_independently_THEN_parameter_unchanged(self, param, param_sp, theta, theta_sp):
```

Comment:
This test is not testing the bug described in the issue. (If you revert the changes in `parameters.py` this will pass anyway)

Instead you need to test the rbv of a `SlitGapParameter` against changes in Theta. `DataMother.beamline_s1_s3_theta_detector` does not contain any `SlitGapParameter`s, only `TrackingPosition`s (i.e. slit heights, not slit gaps). 

You need to write an equivalent method in `DataMother` that constructs a `Beamline` with theta and some slit gap params in it.